src/libraries/System.Drawing.Common/src/System/Drawing/Icon.cs

```diff
@@ -127,7 +127,20 @@ public Icon(Stream stream, int width, int height) : this()
             ArgumentNullException.ThrowIfNull(stream);
 
             _iconData = new byte[(int)stream.Length];
-            stream.Read(_iconData, 0, _iconData.Length);
+#if NET7_0_OR_GREATER
+            stream.ReadExactly(_iconData);
+#else
```

Comment:
Do we want to backport this?

Reply:
I don't think so.  It's been like this forever and to my knowledge no one has complained.  I found it via an audit rather than feedback.